### PR TITLE
Refactor register/finish

### PR DESCRIPTION
**Pull request by nmattia**
This modernizes the code of the registration wrap up, in the style of other new components (like captcha.ts): more structured, more idiomatic `lit` constructs, object parameters and more.

### CODE REVIEW

File: src/frontend/src/flows/register/finish.ts [hunk 1]
```diff
@@ -18,21 +24,28 @@ const pageContent = (userNumber: bigint) => html`
       ${userNumber}
     </div>
     <div class="l-stack">
-      <button id="displayUserContinue" class="c-button">Continue</button>
+      <button
+        @click=${() => onContinue()}
```
Comment (peterpeterparker) [inline src/frontend/src/flows/register/finish.ts:28]: nth: maybe you can even simplify here with `@click=${onContinue}` - not sure, to be tested

File: src/frontend/src/flows/register/finish.ts [hunk 2]
```diff
@@ -18,21 +24,28 @@ const pageContent = (userNumber: bigint) => html`
...
     </div>
   </div>
 `;
 
-export const displayUserNumber = async (userNumber: bigint): Promise<void> => {
-  const container = document.getElementById("pageContent") as HTMLElement;
-  render(pageContent(userNumber), container);
-  return init();
+export const displayUserNumberPage = (
+  props: Parameters<typeof displayUserNumberTemplate>[0],
```
Comment (peterpeterparker) [inline src/frontend/src/flows/register/finish.ts:39]: actually never seen - I think - `[0]` used to express a type, does that work? nice to know

**Review by peterpeterparker — APPROVED** [review]
LGTM, minor comments

File: src/frontend/src/flows/register/finish.ts (hunk 2 above)
Comment (nmattia) [reply src/frontend/src/flows/register/finish.ts:39]: yep, in this case that takes the type of the first param!

Comment (peterpeterparker) [reply src/frontend/src/flows/register/finish.ts:39]: cool

**Pull request merged**